opentelekomcloud/resource_opentelekomcloud_rds_instance_v3.go

```diff
@@ -0,0 +1,1004 @@
+// ----------------------------------------------------------------------------
+//
+//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+//
+// ----------------------------------------------------------------------------
+//
+//     This file is automatically generated by Magic Modules and manual
+//     changes will be clobbered when the file is regenerated.
+//
+//     Please read more about how to change this file at
+//     https://www.github.com/huaweicloud/magic-modules
+//
+// ----------------------------------------------------------------------------
+
+package opentelekomcloud
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/huaweicloud/golangsdk"
+)
+
+func resourceRdsInstanceV3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceRdsInstanceV3Create,
+		Read:   resourceRdsInstanceV3Read,
+		Delete: resourceRdsInstanceV3Delete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"availability_zone": {
```

Comment:
If I understand the docs correct the following format is allowed? 
`availability_zone = "{{ availability_zone_1 }},{{ availability_zone_2 }}"`

I think it would be better to use a array:
`availability_zone = ["{{ availability_zone_1 }}" , "{{ availability_zone_2 }}"]`

Reply:
The api uses the format of "{{ availability_zone_1 }},{{ availability_zone_2 }}", not an array. If creating a single instance ,it should input one availability zone, but for primary/standy instance, it should input two available zones. see the example at https://docs.otc.t-systems.com/en-us/api/rds/en-us_topic_0160329112.html#EN-US_TOPIC_0160329112__table0863181193416

Reply:
You are right the API specifies a string. However I think it would be a better design to specify the availability zones as a array in this TF resource. If we specify the AZs as array you can flatten this array into the string format of the API. A other benefit would be that you can check if the array only contains one AZ if no HA mode is activated and if the array contains two AZs if HA mode is active.

Reply:
agree, I will update.